frontend/workflows/projectSelector/src/selector-reducer.tsx

```diff
@@ -7,6 +7,11 @@ import { Group } from "./types";
 
 const PROJECT_TYPE_URL = "type.googleapis.com/clutch.core.project.v1.Project";
 
+interface DependencyMappings {
+  upstreams?: { [dependency: string]: { [project: string]: boolean } };
+  downstreams?: { [dependency: string]: { [project: string]: boolean } };
+}
```

Comment:
i think we should move these functions and interfaces to a helper file, wdyt?

Reply:
yep sg!